files/unarchive.py

```diff
@@ -89,46 +107,358 @@
 
 import re
 import os
+import stat
+import pwd
+import grp
+import datetime
+import time
+import binascii
 from zipfile import ZipFile
 
 # String from tar that shows the tar contents are different from the
 # filesystem
-DIFFERENCE_RE = re.compile(r': (.*) differs$')
+OWNER_DIFF_RE = re.compile(r': Uid differs$')
+GROUP_DIFF_RE = re.compile(r': Gid differs$')
+MODE_DIFF_RE = re.compile(r': Mode differs$')
+MODE_DIFF_RE = re.compile(r' is newer or same age.$')
```

Comment:
After more testing it seems those changes do fix a problem. gtar --diff compares owner/group information against the content of the archive, which will induce a (false) change when you run it as non-root (or when owner/group is supplied).

A rewrite can properly implement what is needed/expected, similar to what we now do for zip-support.